src/PostgREST/App.hs

```diff
@@ -102,27 +107,28 @@ postgrest logLevel appState connWorker =
   Logger.middleware logLevel Auth.getRole $
     -- fromJust can be used, because the auth middleware will **always** add
     -- some AuthResult to the vault.
-    \req respond -> case fromJust $ Auth.getResult req of
-      Left err -> respond $ Error.errorResponseFor err
-      Right authResult -> do
-        appConf <- AppState.getConfig appState -- the config must be read again because it can reload
-        maybeSchemaCache <- AppState.getSchemaCache appState
-        pgVer <- AppState.getPgVersion appState
-
-        let
-          eitherResponse :: IO (Either Error Wai.Response)
-          eitherResponse =
-            runExceptT $ postgrestResponse appState appConf maybeSchemaCache pgVer authResult req
-
-        response <- either Error.errorResponseFor identity <$> eitherResponse
-        -- Launch the connWorker when the connection is down.  The postgrest
-        -- function can respond successfully (with a stale schema cache) before
-        -- the connWorker is done.
-        when (isServiceUnavailable response) connWorker
-        resp <- do
-          delay <- AppState.getNextDelay appState
-          return $ addRetryHint delay response
-        respond resp
+    \req respond -> inSpanM (getOTelTracer appState) "respond" defaultSpanArguments $
```

Comment:
Hey @develop7!

QQ, this would only grant us otel traces for the JSON error responses right? It would not send any other logs and I think otel is meant to send these as well?

Reply:
No explicit `inSpan*` calls means no traces altogether. https://hackage.haskell.org/package/hs-opentelemetry-instrumentation-auto could help with that, but it requires MonadUnliftIO and, I guess, 

> It would not send any other logs and I think otel is meant to send these as well?

No it won't; not at the moment — while OTel spec _does_ have logs, `hs-opentelemetry` is yet to support them, as well as metrics.